common/lib/plugin_manager.ts

```diff
@@ -63,6 +63,7 @@ class PluginChain<T> {
 
 export class PluginManager {
   private static readonly PLUGIN_CHAIN_CACHE = new Map<[string, HostInfo], PluginChain<any>>();
+  private static readonly GET_HOST_INFO_BY_STRATEGY_PLUGIN_CHAIN_CACHE = new Map<[string, PluginManager], Set<ConnectionPlugin>>();
```

Comment:
Why is the key `[string, PluginManager]`?
Looking at the way this map is accessed, it gets and sets with `[PluginManager.GET_HOST_INFO_BY_STRATEGY_METHOD, this]`, why do we need `PluginManager.GET_HOST_INFO_BY_STRATEGY_METHOD` as part of the key if it never changes?

Reply:
Will `new Map<PluginManager, Set<ConnectionPlugin>>();` work?

Reply:
If we want to use the same map for AcceptsStrategy yes, if we want different ones no

Reply:
Is it possible to have scenarios where acceptsStrategy and and getHostInfoByStrategy use different plugin chains?
Also if we want to use the same map for both lets rename the variable to `getHostInfoByStrategy`

Reply:
On second thought, if it is possible to have scenarios where  acceptsStrategy and getHostInfoByStrategy use different plugin chains, why do we want to use the same cache for both of them instead of just having 2 different caches?